pkg/localstore/gc.go

```diff
@@ -187,13 +177,12 @@ func (db *DB) collectGarbage() (evicted uint64, done bool, err error) {
 
 		totalChunksEvicted++
 
-		i, err := db.retrievalDataIndex.Get(item)
+		storedItem, err := db.retrievalDataIndex.Get(item)
 		if err != nil {
 			return 0, false, err
 		}
-		item.Location = i.Location
 
-		db.metrics.GCStoreTimeStamps.Set(float64(item.StoreTimestamp))
+		db.metrics.GCStoreTimeStamps.Set(float64(storedItem.StoreTimestamp))
 		db.metrics.GCStoreAccessTimeStamps.Set(float64(item.AccessTimestamp))
 
 		// delete from retrieve, pull, gc
```

Comment:
The repetitive error handling can be improved be employing the following:
```Golang
err = nil
deleteInBatch := func(fn func(*leveldb.Batch, shed.Item) error, batch *leveldb.Batch, keyFields shed.Item) {
	if err != nil {
		return
	}
	err = fn(batch, keyFields)
}

deleteInBatch(db.retrievalDataIndex.DeleteInBatch, batch, item)
deleteInBatch(db.retrievalAccessIndex.DeleteInBatch, batch, item)
deleteInBatch(db.pushIndex.DeleteInBatch, batch, storedItem)
deleteInBatch(db.pullIndex.DeleteInBatch, batch, item)
deleteInBatch(db.gcIndex.DeleteInBatch, batch, item)
deleteInBatch(db.postageIndexIndex.DeleteInBatch, batch, storedItem)
deleteInBatch(db.postageChunksIndex.DeleteInBatch, batch, item)
```

Reply:
Interesting. This can be done in a lot of places actually. But I dont want to prolong this change as we need to start testing. Will do this in a follow-up PR later.